source/contributing/documentation.md

```diff
@@ -5,6 +5,45 @@ Documentation contributions should follow to the [writing style recommendations]
 
 Feel free to in touch with the [Nix documentation team](https://nixos.org/community/teams/documentation) if you want to help out.
 
+## Documentation framework
+We aim to build our documentation according to the [Diataxis][diataxis] framework, which divides documentation into four categories:
+
+- Reference
+- How-to
+- Explanation
+- Tutorial
+
+We've found that contributors struggle to know the difference between these categories, while we _strongly_ recommend reading up on the Diataxis framework, we can summarize it in this way:
+
+- Reference
+  - Reference matieral includes technical documentation, API documentation, etc.
+  - Reference material should be relatively terse, simply listing which functions/classes/etc exist, but should still clearly define how to use the code and how it relates to the rest of the ecosystem.
+- How-to (we call these "Recipes")
+  - A recipe is a list of steps to achieve a specific goal.
+  - The goal is not to show how or why something works, but instead to show how to solve a specific problem.
+  - A Recipe assumes that the reader already has the background to understand the topic at hand and therefore doesn't need to explain the introduction of each new concept.
+- Explanation (we call these "Concepts")
+  - A Concept describes the internals of a piece of code or how to think about a particular concept or entity in the ecosystem.
+  - A Concept can also describe the historical context behind why something works the way that it does today.
+  - If you find yourself wanting to write about the nitty gritty details of how something works, you most likely want to write an Explanation.
+- Tutorial
+  - A tutorial walks the user through a particular activity to help give them muscle memory with accepted tools and patterns in the ecosystem.
+  - While the activity itself is important, the goal is to get the reader familiar with the patterns and tools in the ecosystem.
+
+We also find that people struggle with the difference between a How-to/Recipe and a Tutorial.
+One analogy is landing an airplane.
+A person landing a plane for the first time to avoid a crash landing needs a Recipe for how to land the plane, they don't need to know any history or theory, they just need to get the plane on the ground in one piece.
+The documentation they would refer to in this context should simply be a sequence of steps saying "do this, now do that, and, finally, do this last thing."
+An airline pilot in a simulator would refer to a tutorial to walk them through landing an airplane.
+In this context it's fine if the plane crashes, it's just a simulation.
+What's more important is that the airline pilot learns how to land the plane by being clearly walked through the process.
+
+Finally, one last way of thinking about the difference between How-to/Recipe and Tutorial is like this:
+- Recipe: "step 1: do this, step 2: do that, etc"
+- Tutorial: "take my hand as I show you how to do this"
```

Comment:
I don't think this is helpful, both because it's unspecific and because it doesn't alleviate the ambiguity that both types of docs are made of instructions.

Tutorials are a sequence of "do this, then this" just as well, but at a much finer granularity. Their point is to introduce or teach a behavioral pattern, while recipes are just for recalling those patterns.

Here's my take on an intuitive appeal: Tutorials show you how to solve problems you didn't know exist, while recipes remind you of how to solve those problems when you have them.

Reply:
> 
> Here's my take on an intuitive appeal: Tutorials show you how to solve problems you didn't know exist, while recipes remind you of how to solve those problems when you have them.

I might be off on this, but if I interpret your comment correctly @fricklerhandwerk, that would mean that in a tutorial it's okay to run the user into a problem, correct? 

Saying that, we might still decide that it's an antipattern to run users into problems. 
On the other hand if I am driving a  bumper car inside a room with four walls and get taught by four examples to not hit any of those walls, then I have aquired a skill to drive without hitting walls. 🤔 🤷

Reply:
> > Tutorials show you how to solve problems you didn't know exist
>
> would mean that in a tutorial it's okay to run the user into a problem, correct?

No, I meant by that you'd be teaching behavior that allows you solving problems you may not have been aware of before (such as building software with Nix, therefore solving dependency hell). That doesn't mean you have to actually run into issues or tricky boundary conditions while learning the appropriate behavior.